modules/distributor/distributor.go

```diff
@@ -364,21 +364,42 @@ func (d *Distributor) stopping(_ error) error {
 	return services.StopManagerAndAwaitStopped(context.Background(), d.subservices)
 }
 
+// checkForRateLimits checks if the trace batch size exceeds the ingestion rate limit.
+// it will use the ingestion rate limits based on the ingestion strategy configured.
+//
+// LocalIngestionRateStrategy: the ingestion rate limit is applied as is in each distributor.
+// example: if the ingestion rate limit is 10MB/s and the burst size is 20MB, then each distributor
+// will allow 10MB/s with a burst of 20MB.
+//
+// GlobalIngestionRateStrategy: the ingestion rate limit is divided by the number of healthy distributors.
+// example: if the ingestion rate limit is 10MB/s and the burst size is 20MB, and there are 5 healthy distributors,
+// then each distributor will allow 2MB/s with a burst of 20MB.
 func (d *Distributor) checkForRateLimits(tracesSize, spanCount int, userID string) error {
 	now := time.Now()
 	if !d.ingestionRateLimiter.AllowN(now, userID, tracesSize) {
 		overrides.RecordDiscardedSpans(spanCount, reasonRateLimited, userID)
+		// limit: number of bytes per second allowed for the user, as per ingestion rate strategy
 		limit := int(d.ingestionRateLimiter.Limit(now, userID))
+		burst := d.ingestionRateLimiter.Burst(now, userID)
+
+		// globalLimit will be 0 when using local ingestion rate strategy
 		var globalLimit int
 		if d.overrides.IngestionRateStrategy() == overrides.GlobalIngestionRateStrategy {
+			// note: global limit should be calculated using healthy distributors count,
+			// but we are using it in logs, instance count is good enough.
 			globalLimit = limit * d.DistributorRing.InstancesCount()
 		}
+
+		// batch size is too big if it's more than the limit and burst both
+		if tracesSize > limit && tracesSize > burst {
```

Comment:
Is burst always greater than limit?  If so, we can just check burst here, correct?

Reply:
I see a case where burst can be lower then the limit in case you are using global rate limit strategy

limit: 1000, burst 100, but you have 20 distributors but if you are using global limit so per instance limit is 1000/20 = 50, 

but as a rule of thumb per instance limit is usually same or lower then burst.